pkg/controller/controller_worker.go

```diff
@@ -108,6 +108,12 @@ func (c *BundleController) ProcessKey(logger *zap.Logger, key string) (retriable
 		scheme:           c.Scheme,
 		catalog:          c.Catalog,
 	}
+
+	if st.bundle.DeletionTimestamp != nil {
```

Comment:
This check can be done before creating a `bundleSyncTask` instance, but since there is a follow-up PR which will handle the `DeletionTimestamp` in a different way, I would prefer to keep it like that.

Reply:
Should it also be before the building the struct?

Reply:
@amckague which struct are you talking about? if `bundleSyncTask`, have you read my comment? :)

Reply:
the stuct above. no I didn't check it's name